### PR TITLE
 Proxy outputs Eos for end of stream tap events 

### LINKED ISSUE #202: Proxy outputs protocol-specific tap events
This is the proxy side of #198 --- see https://github.com/runconduit/conduit/issues/198#issuecomment-360244903

### PR DESCRIPTION
**Pull request by hawkw**
The proxy now outputs `Eos` instead of `grpc_status` in all end-of-stream tap events. The EOS value is set to `grpc_status_code` when the response ended with a `grpc_status` trailer, `http_reset_code` when the response ended with a reset, and no `Eos` when the response ended gracefully without a `grpc_status` trailer.

This PR updates the proxy. The proto and controller changes are in PR #204.
Part of #198. Closes #202 

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

**Comment by hawkw** [comment]
Not sure how we should resolve merging this with @siggy's PR #204 --- should this be rebased onto #204, or should it wait until #204 is merged and then rebase onto master?

**Comment by hawkw** [comment]
Also, this gets rid of the ugly `pb_response_end` function and turns it into specific `to_tap_event` functions on the different types of end-of-stream events; which has the nice effect of tying certain field values in the tap event directly to the telemetry event it was constructed from. So you can no longer pass arbitrary values to `pb_response_end` and construct weird tap events.

**Comment by siggy** [comment]
@seanmonstar we've left it as a single shared message in common.proto. if we end up needing different things down the road, we can diverge at that point, what do you think?

File: proto/common/common.proto [hunk 1]
```diff
@@ -57,6 +57,14 @@ message Destination {
   string path = 2;
 }
 
+message EosCtx {
```
Comment (olix0r) [inline proto/common/common.proto]: it's still my opinion that it's premature to treat this as common. not a blocker though.

**Comment by hawkw** [comment]
@seanmonstar 
For merging, I'm going to say we merge this into the other PR, since it is required to get the proxy to compile (otherwise the other PR lands a commit that doesn't actually build).

Okay. In that case, it seems like we'd probably want to rebase that PR onto master instead of squashing it, so the proxy changes and controller changes go in separate logical commits?

File: proto/common/common.proto (hunk 1 above)
Comment (siggy) [reply proto/common/common.proto]: @olix0r i'll go ahead and create a duplicate `EosCtx`, but note that one copy will still need to live in common, as that's where `TapEvent` is defined.

**Comment by hawkw** [comment]
Closing this pull request, as the code here will be merged in #204

**Pull request closed without merging**